pkg/apis/build/v1alpha1/buildstrategy.go

```diff
@@ -59,6 +68,31 @@ type Parameter struct {
 	Defaults *[]string `json:"defaults"`
 }
 
+// BuildVolume is a volume that will be mounted in build pod during build step
+// of the Build Strategy
+type BuildVolume struct {
+	// Name of the Build Volume
+	// +required
+	Name string `json:"name"`
+
+	// Description of the Build Volume
+	// +optional
+	Description *string `json:"description,omitempty"`
+
+	// Whether this Volume is optional or not
+	// +optional
+	Optional *bool `json:"optional,omitempty"`
+
+	// Indicates that this Volume can be overrided
+	// in a Build or BuildRun
+	// +optional
+	Overridable *bool `json:"overridable,omitempty"`
+
+	// Represents the source of a volume to mount
+	// +required
+	VolumeSource corev1.VolumeSource `json:"volumeSource"`
```

Comment:
@adambkaplan I need your help here. Why does the ship have the additional `volumeSource` object and the `type` in there?

Reply:
I had my own `VolumeSource` type implemented at first with the `type` field, but after seeing that `TaskSpec` actually has `corev1.VolumeSource` used in it, I decided to use it here, too.
It is possible to use `corev1.VolumeSource` as `inline` in a custom `VolumeSource`, but maintaining all sorts of `type`s for all the options still sees a bit too complex without much reason. I can get it back there though, if needed.

Reply:
Updated and merged in https://github.com/shipwright-io/community/pull/77